### PR TITLE
[BTFS-928]switch status client from http to grpc

### PR DESCRIPTION
**Pull request by laipogo**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Refactoring

* **What is the current behavior?** (You can also link to an open issue here)
send node metrics with http client.

* **What is the new behavior?** (You can also refer to a JIRA ticket here)
send node metrics with grpc client.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
nope

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)
https://github.com/TRON-US/go-btfs-common/pull/15

* **Description of changes**
http client -> grpc client

---

* **Please check if the PR fulfills these requirements**
- [x] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [x] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [x] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [x] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [x] Code changes have run through `go fmt`
- [x] Code changes have run through `go mod tidy`
- [x] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

File: spin/analytics.go [hunk 1]
```diff
@@ -46,27 +50,14 @@ type dataCollection struct {
...
 	log                = logging.Logger("spin")
 	statusServerDomain string
-)
-
-const (
-	routeMetrics = "/metrics"
-	routeHealth  = "/health"
+	callOpts           = []grpc.CallOption{
+		grpc_retry.WithBackoff(grpc_retry.BackoffLinear(100 * time.Millisecond)),
+		grpc_retry.WithMax(3),
```
Comment (wxue) [inline spin/analytics.go]: retry times TBD (no need to adjust now)

**Comment by laipogo** [comment]
> plz target to release branch first, then merge release back to master.
> Code LGTM

got it, will do

File: go.sum [hunk 2]
```diff
@@ -231,6 +235,7 @@ github.com/gorilla/websocket v1.4.0 h1:WDFjx/TMzVgy9VdMMQi2K2Emtwi2QcUQsztZ/zLaH
 github.com/gorilla/websocket v1.4.0/go.mod h1:E7qHFY5m1UJ88s3WnNqhKjPHQ0heANvMoAMk2YaljkQ=
 github.com/gostaticanalysis/analysisutil v0.0.0-20190318220348-4088753ea4d3 h1:JVnpOZS+qxli+rgVl98ILOXVNbW+kb5wcxeGx8ShUIw=
 github.com/gostaticanalysis/analysisutil v0.0.0-20190318220348-4088753ea4d3/go.mod h1:eEOZF4jCKGi+aprrirO9e7WKB3beBRtWgqGunKl6pKE=
+github.com/grpc-ecosystem/go-grpc-middleware v1.0.0 h1:Iju5GlWwrvL6UBg4zJJt3btmonfrMlCDdsejg4CZE7c=
```
Comment (wxue) [inline go.sum]: go tidy ./...

File: spin/analytics.go [hunk 3]
```diff
@@ -184,51 +173,109 @@ func (dc *dataCollection) update(node *core.IpfsNode) {
...
-		dc.reportHealthAlert(fmt.Sprintf("failed to marshal databag: %s", err.Error()))
-		return
+		return err
 	}
+	defer cancel()
+	defer conn.Close()
 
-	// btfs node reports to status server by making HTTP request
-	req, err := http.NewRequest("POST", fmt.Sprintf("%s%s", statusServerDomain, routeMetrics), bytes.NewReader(dataBagMarshaled))
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```
Comment (wxue) [inline spin/analytics.go]: I'd make 5*time.Second a timeout const for future convenient adjustment and avoid the magic number

File: go.sum (hunk 2 above)
Comment (laipogo) [reply go.sum]: actually i do go mod tidy before every `git push` (by bash alias).

**Pull request merged**